node/src/components/consensus/highway_core/state.rs

```diff
@@ -352,6 +353,8 @@ impl<C: Context> State<C> {
             tallies = tallies.filter_descendants(height, bhash, self);
             // If there are no blocks left, `bhash` itself is the fork choice. Otherwise repeat.
             if tallies.is_empty() {
+                let elapsed = start_time.elapsed();
+                trace!(%elapsed,"Time taken for fork-choice to run");
```

Comment:
Not sure if we need to time both this and [`add_valid_vertex`](https://github.com/CasperLabs/casper-node/blob/360945360e3ec6e3682b71c988c74fa8e3923e46/node/src/components/consensus/protocols/highway.rs#L244).

In either case: Can we have a higher resolution than milliseconds? I thought I already saw some other timing measurements elsewhere in the code that had higher accuracy.

Reply:
There are two places where it makes sense to measure how long it takes to compute the forkchoice:
1. When we create vote ourselves and we have to find the parent.
2. When we validate incoming vote and we compute the forkchoice based on vote's panorama.

Reply:
There is a change this replication work, if so, I can close this PR. I shall look for a higher resolution to gauge time

Reply:
[Here are the nanoseconds.](https://github.com/CasperLabs/casper-node/blob/360945360e3ec6e3682b71c988c74fa8e3923e46/node/src/reactor.rs#L408)

@goral09: I don't think it makes sense to measure the run time at the call sites instead of just once, though, and this PR already covers both cases, doesn't it?

Reply:
Sure, this covers both cases. Although, when we were working with Scala node we were seeing big differences in fork-choice times between **1** and **2**. I thought it might be helpful to differentiate from the start.

Reply:
I can't imagine why there would be a difference, though. :confused:

Reply:
Maybe this implementation won't have them. In Scala, we were doing explicit traversals over the j-past-cone of the message.

Reply:
I've updated it to track in nano-seconds

Reply:
Are we going to remove the measurement in [`add_valid_vertex`](https://github.com/CasperLabs/casper-node/blob/360945360e3ec6e3682b71c988c74fa8e3923e46/node/src/components/consensus/protocols/highway.rs#L244) ?

Reply:
I'd argue its out of scope for this PR?

Reply:
But it would just remove three lines of code, wouldn't it? I'd also be in favor of removing the other measurement, especially since it's less precise.

Reply:
Can you remove the tracing from `add_valid_vertex` @darthsiroftardis ?

Reply:
Can do!